model/grader.py

```diff
@@ -0,0 +1,69 @@
+from .line_segment import LineSegment
+from evaluators.interfaces import LineSegmentEvaluator
+from model.serializer import Serializer
+import json
+from texttable import Texttable
+
+
+class Grader:
+    def __init__(self, evaluator_type: LineSegmentEvaluator, ideal_solution_path: str, predicted_solution_path: str) -> None:
+        self.evalutor = evaluator_type
+        serializer = Serializer()
+        ideal_line_segments: list[dict[str, list[LineSegment]]] = serializer.to_line_segment(self._load_json(ideal_solution_path))
```

Comment:
doesn't this need to be in self.ideal_line_segments? Otherwise how is it referenced in other methods? It's not self encapsulating and becomes a global if you do this.

Reply:
It's not referenced in other methods as of now. Might change it if am going to use it in other methods

Reply:
oh i see, i misread it.

I think either way, we shouldn't be calling self.calculate_distances here. See my other comment below. For now i'll resolve this but if you could address the other comment, that'd be great.